src/udev/udev-builtin-input_id.c

```diff
@@ -240,8 +240,12 @@ static bool test_pointers(struct udev_device *dev,
                         is_touchscreen = true;
         }
 
-        if (has_rel_coordinates && has_mouse_button)
-                is_mouse = true;
+        if (has_mouse_button) {
+                if (has_rel_coordinates)
+                        is_mouse = true;
+                else if (!has_abs_coordinates) /* mouse buttons and no axis */
+                        is_mouse = true;
```

Comment:
Also a note on the commit message: the first sentence says "Assign input class...", but I think it'd be clearer to be explicit and say "Label devices which have mouse buttons and no axis as ID_INPUT_MOUSE=1.".